ReactNative/ReactTwitter/core/views/main-navigator.js

```
@@ -64,28 +68,15 @@ var NavigationBarRouteMapper = {
   },
 
   _getTitleByScreenId (screenId) {
-    switch (screenId) {
-      case splashScreenID:
-        return 'SplashScreen'
-      case loginScreenID:
-        return 'Login'
-      case debugScreenID:
-        return 'Debug Screen'
-      case deepLinkingID:
-        return 'Deep Linking'
-      case pushNotificationsScreenID:
-        return 'Push Notifications'
-      case OauthViewID:
-        return 'Oauth'
-      case tweetsListID:
-        return 'Timeline'
-      case tweetViewID:
-        return 'Tweet Detail'
-      case debugAnimationSceneID:
-        return 'Debug Animation'
-      default:
-        return ''
-    }
+    let title = ''
+    scenes.forEach((scene) => {
+      if (scene.navigatorID() === screenId) {
+        title = scene.navigatorTitle()
+        return
+      }
+    })
+
+    return title
```

I would Throw an exception here: fail fast